test/e2e/cruise/unoffload/unoffload_test.go

```diff
@@ -0,0 +1,261 @@
+		var pod *corev1.Pod
+		Eventually(func() corev1.PodStatus {
+			pods, err := util.GetPodsFromDeployment(ctx, cluster.ControllerClient, namespaceName, deploymentName)
+			if err != nil {
+				return corev1.PodStatus{}
+			}
+
+			switch len(pods) {
+			case 0:
+				return corev1.PodStatus{}
+			case 1:
+				pod = &pods[0]
+				return pod.Status
+			default:
+				Fail(fmt.Sprintf("found more than one pod for deployment %s", deploymentName))
+				return corev1.PodStatus{}
+			}
+		}, timeout, interval).Should(MatchFields(IgnoreExtras, Fields{
+			"Phase": Equal(corev1.PodRunning),
+		}))
+
+		// check the pod in the remote cluster
+		podNode := pod.Spec.NodeName
+		var node corev1.Node
+		Expect(cluster.ControllerClient.Get(ctx, client.ObjectKey{Name: podNode}, &node)).To(Succeed())
+
+		clusterID, ok := node.Labels[consts.RemoteClusterID]
+		Expect(ok).To(BeTrue())
+		Expect(clusterID).ToNot(BeEmpty())
+
+		var testerIndex = -1
+		for i := range testContext.Clusters {
+			if i == homeIndex {
+				continue
+			}
+
+			remoteClusterID, err := utils.GetClusterID(ctx, testContext.Clusters[i].NativeClient, "liqo")
+			Expect(err).ToNot(HaveOccurred())
+			if string(remoteClusterID) == clusterID {
+				testerIndex = i
+				break
+			}
+		}
+		Expect(testerIndex).ToNot(Equal(-1))
+
+		// check the pod
+		Eventually(func() []corev1.PodStatus {
+			var podList corev1.PodList
+			_ = testContext.Clusters[testerIndex].ControllerClient.List(ctx, &podList, client.InNamespace(namespaceName))
+			return sliceutils.Map(podList.Items, func(p corev1.Pod) corev1.PodStatus {
+				return p.Status
+			})
+		}, timeout, interval).Should(ContainElement(MatchFields(IgnoreExtras, Fields{
+			"Phase": Equal(corev1.PodRunning),
+		})))
+
+		// check the service
+		for i := range testContext.Clusters {
+			if testContext.Clusters[i].Role == liqov1alpha1.ConsumerRole {
+				continue
+			}
+
+			Expect(testContext.Clusters[i].ControllerClient.Get(ctx,
+				client.ObjectKey{Name: deploymentName, Namespace: namespaceName}, &corev1.Service{})).To(Succeed())
+		}
+
+		// check the config map
+		for i := range testContext.Clusters {
+			if testContext.Clusters[i].Role == liqov1alpha1.ConsumerRole {
+				continue
+			}
+
+			Expect(testContext.Clusters[i].ControllerClient.Get(ctx,
+				client.ObjectKey{Name: "test-configmap", Namespace: namespaceName}, &corev1.ConfigMap{})).To(Succeed())
+		}
+	}
+)
+
+var _ = Describe("Liqo E2E", func() {
+	Context("Unoffload", func() {
+
+		BeforeEach(func() {
+			// ensure the namespace is created
+			for i := range testContext.Clusters {
+				if testContext.Clusters[i].Role == liqov1alpha1.ConsumerRole {
+					// ensure the namespace is created
+					Expect(util.Second(util.EnforceNamespace(ctx, testContext.Clusters[i].NativeClient,
+						testContext.Clusters[i].Cluster, namespaceName))).To(Succeed())
+
+					_ = util.OffloadNamespace(testContext.Clusters[i].KubeconfigPath, namespaceName,
+						"--namespace-mapping-strategy=EnforceSameName")
```

Comment:
Why the error is discarded?

Reply:
Because if one test fails, the namespace may already be offloaded, and the function returns errors if the namespace is already offloaded so that the next test will fail